packages/hawtio/src/plugins/camel/routes/visualization-service.ts

```diff
@@ -0,0 +1,220 @@
+            if (componentScheme) {
+              // const value = routesService.getIcon(componentScheme)
+              // if (value) {
+              //   imageUrl = Core.url(value)
+              // }
+            }
+          }
+        }
+
+        let cid = route.getAttribute('_cid') || route.getAttribute('id')
+        node = {
+          id: id,
+          name: nodeId,
+          label: label,
+          labelSummary: labelSummary,
+          group: 1,
+          elementId: elementID,
+          imageUrl: imageUrl,
+          cid: cid ?? id,
+          tooltip: tooltip,
+          type: nodeId,
+          uri: uri ?? '',
+        }
+        if (rid) {
+          node.cid = rid
+        }
+        if (!cid) {
+          cid = nodeId + (nodes.length + 1)
+        }
+        if (cid) {
+          node.cid = cid
+        }
+        // only use the route id on the first from node
+        rid = null
+        nodes.push(node)
+        if (parentId !== null && parentId !== id) {
+          if (siblingNodes.length === 0 || parenNodeName === 'choice') {
+            links.push({ id: parentId + '-' + id, source: parentId + '', target: id, type: edgeType })
+          } else {
+            siblingNodes.forEach(function (nodeId) {
+              links.push({ id: nodeId + '-' + id, source: nodeId + '', target: id, type: edgeType })
+            })
+            siblingNodes.length = 0
+          }
+        }
+      } else {
+        // ignore non EIP nodes, though we should add expressions...
+        const langSettings = { name: 'sfd' } // Camel.camelLanguageSettings(nodeId)
+        if (langSettings && parentNode) {
+          // lets add the language kind
+          const name = langSettings['name'] || nodeId
+          const text = route.textContent
+
+          if (text) {
+            parentNode.tooltip = parentNode.label + ' ' + name + ' ' + text
+            parentNode.label += ': ' + this.appendLabel(route, text, true)
+          } else {
+            parentNode.label += ': ' + this.appendLabel(route, name, false)
+          }
+        }
+      }
+
+      const siblings = this.addRouteXmlChildren(route, nodes, links, id, node)
+      if (parenNodeName === 'choice') {
+        siblingNodes = siblingNodes.concat(siblings)
+      } else if (
+        nodeId === 'aggregate' ||
+        nodeId === 'choice' ||
+        nodeId === 'delay' ||
+        nodeId === 'filter' ||
+        nodeId === 'loadBalance' ||
+        nodeId === 'loop' ||
+        nodeId === 'multicast' ||
+        nodeId === 'resequence' ||
+        nodeId === 'split' ||
+        nodeId === 'doTry' ||
+        nodeId === 'doCatch' ||
+        nodeId === 'doFinally' ||
+        nodeId === 'idempotentConsumer' ||
+        nodeId === 'onCompletion'
+      ) {
+        siblingNodes = siblings
+      } else {
+        siblingNodes = [nodes.length - 1]
+      }
+    }
+    return siblingNodes
+  }
+}
+export const routeVizualisationService = new VisualizationService()
```

Comment:
`routeVizualisationService` -> `routeVisualizationService` :-)
Or you can maybe simply name `visualizationService`?